spanner/session.go

```diff
@@ -72,7 +72,9 @@ func (sh *sessionHandle) recycle() {
 	}
 	p := sh.session.pool
 	tracked := sh.trackedSessionHandle
+	sh.mu.Unlock()
 	sh.session.recycle()
```

Comment:
I might be wrong, but is there any possibility that [sessionHandle.destory()](https://github.com/googleapis/google-cloud-go/blob/spanner/v1.30.0/spanner/session.go#L135) is called during `sh.session.recycle()` is being called?

Reply:
No, as per my understanding destroy is only called when isSessionNotFoundError is returned from server.

Reply:
No, I was just wondering if it's ok to release a lock during `sh.session.recycle()` call. It looks good to me.

Reply:
I think it would be safer and clearer if you would just store `sh.session` in a local variable `s` while the lock is still taken, and then after (original) line 80, call `s.recycle()`. So you get something like:

```go
	p := sh.session.pool
	tracked := sh.trackedSessionHandle
	s := sh.session
	sh.session = nil
	sh.trackedSessionHandle = nil
	sh.checkoutTime = time.Time{}
	sh.stack = nil
	sh.mu.Unlock()
        s.recycle()
```